WIP: Updating action versions

algojack opened the pull request:
testing...

algorandskiy commented:
> updated tools package version to match golangci-lint (root cause)

can we add some checks to ensure `golangci-lint` build matches to the plugin (part test linter)? Or, I guess why was  `golangci-lint` updated without noticing?

algobarb reviewed (approved):
The code changes relatively seems similar to what was in the PR summary, so I think the changes look good if they've been tested.

algojack commented:
> > updated tools package version to match golangci-lint (root cause)
> 
> can we add some checks to ensure `golangci-lint` build matches to the plugin (part test linter)? Or, I guess why was `golangci-lint` updated without noticing?

Added a fix to error out properly if it doesn't run: broke up the piped commands into two and added set -e and tested on the bugs we had here.

It was updated without noticing because it failed silently (without failing the job)

algojack commented on .github/workflows/reviewdog.yml:
```diff
@@ -80,17 +80,21 @@ jobs:
+          -name="Lint Warnings" \
+          -reporter=github-check \
+          -filter-mode=added \
+          -fail-on-error=true \
```
i believe they are warnings, not failures, but i'll see if we can display these messages on another page too

algojack replied:
i'd think that this is an issue for reviewdog to handle, but we can pipe it into GITHUB_STEP_SUMMARY or GITHUB_OUTPUT for example. https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary
